src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeCache.targets

```diff
@@ -23,10 +23,120 @@ Copyright (c) .NET Foundation. All rights reserved.
+    
+    <ItemGroup>
+      <_AllProjects Include="$(AdditionalProjects.Split('%3B'))"/>
+      <_AllProjects Include ="$(MSBuildProjectFullPath)"/>
+    </ItemGroup>
+    
+    <MSBuild Projects="%(_AllProjects.Identity)"
+                 Targets="CacheWorkerMapper"
+                 Properties="ComposeWorkingDir=$(ComposeWorkingDir);
+                             PublishDir=$(PublishDir);
+                             TargetFramework=$(_TFM);
+                             JitPath=$(JitPath);
+                             Crossgen=$(Crossgen);
+                             DisableImplicitFrameworkReferences=true;
+                             SkipUnchangedFiles=$(SkipUnchangedFiles);
+                             PreserveCacheLayout=$(PreserveCacheLayout)"
+                  ContinueOnError="WarnAndContinue"  />
+   </Target>
+  <!--
+    ============================================================
+                                        CacheWorkerMapper
+
+   Spins of the compose mappers
+    ============================================================
+    -->
+
+  <Target Name="CacheWorkerMapper">
+    
+    <MSBuild Projects="$(MSBuildProjectFullPath)"
+                 Targets="CacheWorker"
+                 Properties="CachePackageName=%(PackageReference.Identity);
+                             CachePackageVersion=%(PackageReference.Version);
+                             ComposeWorkingDir=$(ComposeWorkingDir);
+                             PublishDir=$(PublishDir);
+                             TargetFramework=$(TargetFramework);
+                             JitPath=$(JitPath);
+                             Crossgen=$(Crossgen);
+                             SkipUnchangedFiles=$(SkipUnchangedFiles);
+                             PreserveCacheLayout=$(PreserveCacheLayout)"
+                  ContinueOnError="WarnAndContinue"
+             />
+  </Target>
+
+  <Target Name="CacheWorker"
+          DependsOnTargets="_InitializeBasicProps;
+                            PrepforRestoreForComposeCache;
+                            CacheWorkerPerformWork"/>
   
+  <Target Name="CacheWorkerPerformWork"
+          DependsOnTargets="RestoreForComposeCache;
+                            ComputeAndCopyFilesToCacheDirectory;"
+          Condition="!Exists($(CacheWorkerWorkingDir))" />
+
+<!--
+    ============================================================
+                                        CacheFinalizer
+
+   Cleansup and produces artifacts after Completion of cache
+    ============================================================
+    -->
+  
+  <Target Name="CacheFinalizer">
+    
+    <ItemGroup>
+      <ListOfPackageReference Include="@(AllResolvedPackagesPublished -> '%20%20%20%20%20&lt;PackageReference Include=&quot;%(Identity)&quot;  Version =&quot;%(Version)&quot;/&gt;')"/>
+    </ItemGroup>
+    <PropertyGroup>
+     <_CacheArtifactContent>
+      <![CDATA[
+<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="$(MSBuildToolsVersion)">
```

Comment:
If you insist on using MSBuild format - there is no need putting `ToolsVersion` in the file.  We don't in any of our templates.